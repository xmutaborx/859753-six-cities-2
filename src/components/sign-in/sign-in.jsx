import React from 'react';
import PropTypes from 'prop-types';
import Operation from '../../store/operation';
import {connect} from 'react-redux';

const SignIn = (props) => {
  let loginInput;
  let password;

  const handleSubmit = (event) => {
    event.preventDefault();
    props.submitForm(loginInput.value, password.value);
  };

  return (
    <div className="page page--gray page--login">
      <main className="page__main page__main--login">
        <div className="page__login-container container">
          <section className="login">
            <h1 className="login__title">Sign in</h1>
            <form
              className="login__form form"
              action="#"
              method="post"
              onSubmit={(e) => handleSubmit(e)}
            >
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">E-mail</label>
                <input
                  className="login__input form__input"
                  type="email"
                  name="email"
                  placeholder="Email"
                  required
                  ref={(e) => {
                    loginInput = e;
                  }}
                />
              </div>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">Password</label>
                <input
                  className="login__input form__input"
                  type="password"
                  name="password"
                  placeholder="Password"
                  required
                  ref={(e) => {
                    password = e;
                  }}
                />
              </div>
              <button className="login__submit form__submit button" type="submit">Sign in</button>
            </form>
          </section>
          <section className="locations locations--login locations--current">
            <div className="locations__item">
              <a className="locations__item-link" href="#">
                <span>Amsterdam</span>
              </a>
            </div>
          </section>
        </div>
      </main>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  submitForm: (login, password) => dispatch(Operation.authorization(login, password)),
});

SignIn.propTypes = {
  submitForm: PropTypes.func.isRequired,
};

export {SignIn};
export default connect(null, mapDispatchToProps)(SignIn);
