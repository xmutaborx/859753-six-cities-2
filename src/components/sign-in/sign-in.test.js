import React from 'react';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16/build";
import toJson from "enzyme-to-json";

import {SignIn} from './sign-in.jsx';

Enzyme.configure({adapter: new Adapter()});

it(`SignIn Page correctly renders`, () => {
  const tree = shallow(<SignIn
    onSubmitForm={() => {}}
    onChangeLogin={() => {}}
    onChangePassword={() => {}}
    login={`Login`}
    password={`Password`}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
