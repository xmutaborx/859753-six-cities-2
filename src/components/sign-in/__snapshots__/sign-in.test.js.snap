// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers Page correctly renders 1`] = `
<div
  className="page page--gray page--login"
>
  <main
    className="page__main page__main--login"
  >
    <div
      className="page__login-container container"
    >
      <section
        className="login"
      >
        <h1
          className="login__title"
        >
          Sign in
        </h1>
        <form
          action="#"
          className="login__form form"
          method="post"
          onSubmit={[Function]}
        >
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              E-mail
            </label>
            <input
              className="login__input form__input"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
              value="Login"
            />
          </div>
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              Password
            </label>
            <input
              className="login__input form__input"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
              value="Password"
            />
          </div>
          <button
            className="login__submit form__submit button"
            type="submit"
          >
            Sign in
          </button>
        </form>
      </section>
      <section
        className="locations locations--login locations--current"
      >
        <div
          className="locations__item"
        >
          <Link
            className="locations__item-link"
            to="/"
          >
            <span>
              Amsterdam
            </span>
          </Link>
        </div>
      </section>
    </div>
  </main>
</div>
`;
